internal/test/testdata.go

```diff
@@ -209,5 +209,5 @@ func MachineDeployment(opts ...MachineDeploymentOpt) *clusterv1.MachineDeploymen
 		opt(md)
 	}
 
-	return md
+	return md.DeepCopy()
```

Comment:
why deep copy?

Reply:
Ah, I was suppose remove this from here, I was testing something here. Thanks for catching that